### DISCUSSION TITLE
Upgrade scenarios

### DISCUSSION THREAD

**Discussion by kentrix70** [cupcakearmy/cryptgeon#82] [Q&A] [answered]
Hi there.
Just a few questions regarding upgrade of servers with docker running the tool and also upgrading of Cryptgeon itself.
When more and more people are using this tool internally in a large organization, it becomes important to plan upgrades.
As of now it is not possible to prevent "flush" of live secure notes sent but not yet received, when the container is restarted.

We thought about running 2 instances and load balance them, but the problem is, that as the notes are in memory only, if we switch the instance, notes that are sent before the switching took place can not be delivered. And as secure notes are generated quite often, this is becoming an issue.
Is there any way you can "dump" the current secure note repo to an encrypted file or transfer of secure notes from one instance to another? Any other thoughts or recommendations on this?

**Comment by cupcakearmy** [comment, marked as answer]
Sure, just use the redis persistence option :)
https://redis.io/docs/management/persistence/

For load balancing, you would need to use redis cluster, they also support it. The backend of cryptgeon is stateless
